sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -299,27 +415,46 @@ public void getTempFilePrefix(ProcessContext c) {
                             false,
                             bigQueryServices,
                             jobIdTokenView,
-                            schemasView,
                             WriteDisposition.WRITE_EMPTY,
                             CreateDisposition.CREATE_IF_NEEDED,
                             dynamicDestinations))
                     .withSideInputs(writeTablesSideInputs));
 
-    // This view maps each final table destination to the set of temporary partitioned tables
-    // the PCollection was loaded into.
-    PCollectionView<Map<TableDestination, Iterable<String>>> tempTablesView =
-        tempTables.apply("TempTablesView", View.<TableDestination, String>asMultimap());
-
-    singleton.apply(
-        "WriteRename",
-        ParDo.of(
-                new WriteRename(
-                    bigQueryServices,
-                    jobIdTokenView,
-                    writeDisposition,
-                    createDisposition,
-                    tempTablesView))
-            .withSideInputs(tempTablesView, jobIdTokenView));
+    if (triggeringFrequency == null) {
```

Comment:
This method has grown to be several hundred lines long and has 3 branches on triggeringFrequency. I recommend refactoring it into two methods, for triggered and untriggered versions, reusing the common code via private utility functions. (refactoring into more sub-transforms may or may not be a good idea)